code/datums/gamemode/objectives/target/target.dm

```diff
@@ -22,20 +15,31 @@
 /datum/objective/target/PostAppend()
 	if(auto_target)
 		return find_target()
-	return TRUE
-
-/datum/objective/target/delayed/PostAppend()
-	if(emergency_shuttle.location || emergency_shuttle.direction == 2)
+	if(delay && (emergency_shuttle.location || emergency_shuttle.direction == 2))
 		PostDelay() //If the shuttle is docked or en route to centcomm, no delay
 		return TRUE
 	spawn(delay)
 		PostDelay()
 	return TRUE
 
+/datum/objective/target/ShuttleDocked(state)
+	if (delay && !target && (state == 1))
+		PostDelay()
+
+/datum/objective/target/proc/PostDelay() // reveals the target
+	target = delayed_target
+	explanation_text = format_explanation()
+	owner.current.playsound_local(owner.current.loc, 'sound/machines/twobeep.ogg', 75, 0)
+	to_chat(owner.current, "<span class='userdanger'>Target Revealed</span>: <b>[explanation_text]</b>")
+
 /datum/objective/target/proc/find_target()
 	var/list/possible_targets = get_targets()
 	if(possible_targets.len > 0)
-		target = pick(possible_targets)
+		if (delay)
+			delayed_target = pick(possible_targets)
+			to_chat(owner.current, "<span class='danger'>Your target's identity will be revealed to you in [delay/600] MINUTE[(delay > 600) ? "S":""].</span>")
```

Comment:
No round? Regardless we may have a helper to turn this into minutes and seconds remaining.

Reply:
Nevermind, this isn't decrementing, duh